### PR TITLE
fix: Use fully qualified clang executable path instead of "clang"

### PR DESCRIPTION
**Pull request by DerekMaggio**

### CODE REVIEW

File: FindClang.cmake [hunk 1]
```diff
@@ -80,6 +80,14 @@ if (Clang_CODECHECKER_EXECUTABLE STREQUAL "Clang_CODECHECKER_EXECUTABLE_NOTFOUND
   message(WARNING "Could not find codechecker, which is system-dependent. See https://codechecker.readthedocs.io/en/latest/#install-guide")
 endif()
 
+# Need to add symlink to executable because command `clang` does not exist
+if(${CMAKE_HOST_SYSTEM_NAME} STREQUAL "Linux")
+  execute_process(
+    WORKING_DIRECTORY /usr/bin
```
Comment (sotaoverride) [inline FindClang.cmake:86]: ln command can be run from anywhere (think its added to $PATH). maybe stay out of /usr/bin and just give complete paths in ln command

Comment (sotaoverride) [reply FindClang.cmake:86]: should we be checking if clang bin does already exist?

Comment (DerekMaggio) [reply FindClang.cmake]: Instead of doing this. I am just specifying the correct executable. See my subsequent commit

**Review by sfoster1 — APPROVED** [review]
Nice, LGTM

**Review by Laura-Danielle — APPROVED** [review]
nice catch

File: FindClang.cmake [hunk 2]
```diff
@@ -82,7 +82,7 @@ endif()
 
 if(NOT Clang_EXECUTABLE STREQUAL "Clang_EXECUTABLE-NOTFOUND")
   execute_process(
-    COMMAND clang --version
+    COMMAND ${Clang_EXECUTABLE} --version
```
Comment (sotaoverride) [inline FindClang.cmake:85]: you call find_program(Clang_EXECUTABLE
  clang ... earlier, so what difference does it make if you use clang / Clang_EXECUTABLE here.. am i missing something?

Comment (DerekMaggio) [reply FindClang.cmake:85]: `find_program` searches for the program and stores it's value as the first variable passed to `find_program`
So in the case of, 
```
find_program(Clang_EXECUTABLE
  clang
  PATHS ${CMAKE_SOURCE_DIR}/stm32-tools/clang/${CMAKE_HOST_SYSTEM_NAME}
  PATH_SUFFIXES bin)
```
We are searching for clang inside `ot3-firmware/stm32-tools/clang/Linux` and storing it to the `Clang_EXECUTABLE` variable. 

I am assuming that on Mac the clang executable is in fact just `clang` whereas on my Linux system it is `ot3-firmware/stm32-tools/clang/Linux/clang`

**Pull request merged**